src/cli/commands/test/iac-local-execution/parsers.ts

```diff
@@ -0,0 +1,65 @@
+import * as hclToJson from 'hcl-to-json';
+import * as YAML from 'js-yaml';
+import { EngineType, IacFileData, IacFileMetadata } from './types';
+
+const REQUIRED_K8S_FIELDS = ['apiVersion', 'kind', 'metadata'];
+
+export function tryParseIacFile(
+  fileMetadata: IacFileMetadata,
+  fileContent: string,
+): Array<IacFileData> {
+  switch (fileMetadata.fileType) {
+    case 'yaml':
+    case 'yml':
+    case 'json':
+      return tryParsingKubernetesFile(fileContent, fileMetadata);
+    case 'tf':
+      return [tryParsingTerraformFile(fileContent, fileMetadata)];
+    default:
+      throw new Error('Invalid IaC file');
```

Comment:
Could have a more descriptive message e.g.

```suggestion
      throw new Error(`Unsupported filetype ${fileMetadata.fileType }`);
```

Reply:
going to do a bit of an overhaul to the error handling in the next pr for Dir-Support, will take care of it over there.
tnx 👍